helix-rest/src/test/java/org/apache/helix/rest/server/TestMSDAccessorLeaderElection.java

```diff
@@ -99,7 +99,9 @@ public void beforeClass() throws Exception {
 
     // Set the new uri to be used in leader election
     System.setProperty(MetadataStoreRoutingConstants.MSDS_SERVER_HOSTNAME_KEY,
-        getBaseUri().getHost() + ":" + newPort);
+        getBaseUri().getHost());
+    System
+        .setProperty(MetadataStoreRoutingConstants.MSDS_SERVER_PORT_KEY, Integer.toString(newPort));
```

Comment:
Do these not need to be cleared after the test?

Reply:
Nope, the test base it extends clears up the system properties. These 2 lines are overwriting the old properties set by the test base, and the test base cleans up after itself.